flask/common/src/main/java/net/corda/flask/common/Flask.java

```diff
@@ -0,0 +1,111 @@
+package net.corda.flask.common;
+
+import lombok.SneakyThrows;
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.math.BigInteger;
+import java.security.DigestInputStream;
+import java.security.MessageDigest;
+import java.util.AbstractMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.zip.CRC32;
+import java.util.zip.ZipEntry;
+
+public class Flask {
+
+    public static class Constants {
+        public static final String DEFAULT_LAUNCHER_NAME = "net.corda.flask.launcher.Launcher";
+        public static final String LIBRARIES_FOLDER = "LIB-INF";
+        public static final String METADATA_FOLDER = "META-INF";
+        public static final int BUFFER_SIZE = 0x10000;
+    }
+
+    public static class ManifestAttributes {
+        public static final String APPLICATION_CLASS = "Application-Class";
+        public static final String JVM_ARGS = "JVM-Args";
+        public static final String JAVA_AGENTS = "Java-Agents";
+        public static final String ENTRY_HASH = "MD5-Digest";
```

Comment:
MD5 is a broken hash algorithm. Who is requiring it? I think the default hash algorithm these days is SHA-256.

Reply:
Yes nothing in R3 should use MD5 or SHA-1 it is trivial to get collisions these days and it flags red in due diligence audits.

Reply:
Using truncated SHA2-256 is the usual approach if we need to have limited length (though 128-bit is minimum length to avoid collisions)

Reply:
there is no length limit, I can happily use SHA-256 or even SHA-512, the reason I originally chose MD5 is because I thought it was not a security critical part of the code (it is only used to deduplicate libraries in the cache folder avoiding to extract them again if they are already there). But I see there could be security issues if one is able to replace one of our libraries in the cache with a different jar with the same MD5 hash, I'll upgrade the hashing algorithm

Reply:
I replaced MD5 with SHA-256